spec/factories/report.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+def test_result_score(counter, threshold)
+  threshold + (counter.zero? ? -1 : 1) * (0..(100 - threshold)).to_a.sample
+end
```

Comment:
You can push this "logic" into the `test_result` factory and not have it as a method.